src/Microsoft.AspNet.Hosting/Internal/HostingLoggerExtensions.cs

```diff
@@ -0,0 +1,135 @@
+            }
+        }
+
+        private class HostingRequestScope : ILogValues
+        {
+            private readonly HttpContext _httpContext;
+
+            private FeatureReference<IHttpRequestIdentifierFeature> _requestIdentifierFeatureReference;
+            private IEnumerable<KeyValuePair<string, object>> _cachedGetValues;
+            private string _cachedToString;
+
+            public HostingRequestScope(HttpContext httpContext)
+            {
+                this._httpContext = httpContext;
+            }
+
+            public IHttpRequestIdentifierFeature RequestIdFeature =>
+                _requestIdentifierFeatureReference.Fetch(_httpContext.Features) ??
+                _requestIdentifierFeatureReference.Update(_httpContext.Features, new FastHttpRequestIdentifierFeature());
+
+            public override string ToString() => _cachedToString ?? Interlocked.CompareExchange(
+                ref _cachedToString,
+                $"RequestId:{RequestIdFeature.TraceIdentifier} RequestPath:{_httpContext.Request.Path}",
+                null);
+
+            public IEnumerable<KeyValuePair<string, object>> GetValues() => _cachedGetValues ?? Interlocked.CompareExchange(
+                ref _cachedGetValues,
+                new[]
+                {
+                    new KeyValuePair<string, object>("RequestId", RequestIdFeature.TraceIdentifier),
+                    new KeyValuePair<string, object>("RequestPath", _httpContext.Request.Path.ToString()),
+                },
+                null);
+        }
+
+        private class HostingRequestStarting : ILogValues
+        {
+            internal static readonly Func<object, Exception, string> Callback = (state, exception) => ((HostingRequestStarting)state).ToString();
+
+            private readonly HttpContext _httpContext;
+            private IEnumerable<KeyValuePair<string, object>> _cachedGetValues;
+            private string _cachedToString;
+
+            public HostingRequestStarting(HttpContext httpContext)
+            {
+                _httpContext = httpContext;
+            }
+
+            public override string ToString() => _cachedToString ?? Interlocked.CompareExchange(
+                ref _cachedToString,
+                $"Request starting {_httpContext.Request.Protocol} {_httpContext.Request.Method} {_httpContext.Request.Scheme}://{_httpContext.Request.Host}{_httpContext.Request.PathBase}{_httpContext.Request.Path}{_httpContext.Request.QueryString} {_httpContext.Request.ContentType} {_httpContext.Request.ContentLength}",
+                null);
+
+            public IEnumerable<KeyValuePair<string, object>> GetValues() => _cachedGetValues ?? Interlocked.CompareExchange(
+                ref _cachedGetValues,
+                new[]
+                {
+                    new KeyValuePair<string, object>("EventName", "RequestStarting"),
```

Comment:
Scopes are not log messages, they're one or two properties added as decoration to log messages inside their active range. So adding the same EventName to every message in the scope's range isn't good.